experiments/testers/cc/create_samples_mean.cc

```diff
@@ -0,0 +1,258 @@
+// Additional parameters enable large values to be added
+// to the algorithm one by one.
+double DPLargeMean(double initial_value, double extra_values_length, 
+  double extra_value, double granularity, double epsilon, int max_partitions,
+  int max_contributions, int lower, int upper) {
+    std::unique_ptr<BoundedMean<double>> boundedmean =
+    BoundedMean<double>::Builder()
+    .SetEpsilon(epsilon)
+    .SetMaxPartitionsContributed(max_partitions)
+    .SetMaxContributionsPerPartition(max_contributions)
+    .SetLower(lower)
+    .SetUpper(upper) 
+    .Build()
+    .ValueOrDie();
+
+// Add entry with initial value
+  boundedmean->AddEntry(DiscretizeMean(initial_value,granularity));
+
+// Add entry with subsequent values
+  for (int i=0; i<extra_values_length; i++) {
+    boundedmean->AddEntry(extra_value);
+  }
+  return GetValue<double>(boundedmean->PartialResult().ValueOrDie());
+}
+
+  // std::vector<double> discretized_values;
+  // for (double i : values) {
+  //   double discretized_value = DiscretizeMean(i, granularity);
+  //   discretized_values.push_back(discretized_value);
+  // }
+
+// Creates a folder to contain all samples with a particular ratio value
+// (e.g., R95). Every folder contains 22 subfolders for each unique sample-pair.
+// Every subfolder contains seven runs of each sample-pair (14 files in total).
+void CreateSingleScenarioMean(int scenario, std::vector<double>valuesA,
+  std::vector<double>valuesB, double granularity, double epsilon,
+  int max_partitions, int max_contributions, int lower, int upper,
+  int number_of_samples, double ratio, double initial_value,
+  double extra_values_length, double extra_value) { 
+
+  double implemented_epsilon = epsilon / ratio;
+  std::string filepath = mean_samples_folder+"/R"
+    +std::to_string(static_cast<int>(ratio*100))+"/Scenario"
+    +std::to_string(scenario);
+  mkdir(filepath.c_str(), 0777);
+
+  for (int i=0; i<7; i++) {
+    std::ofstream samplefileA;
+    std::ofstream samplefileB;
+    samplefileA.open(filepath+"/TestCase"+std::to_string(i)+"A.txt");
+    samplefileB.open(filepath+"/TestCase"+std::to_string(i)+"B.txt");
+
+    if (extra_values_length == 0) {
+      for (int i=0; i<number_of_samples; i++) {
+        double outputA = DPMean(valuesA, granularity,
+          implemented_epsilon, max_partitions, max_contributions, lower, upper);
+        double discretized_outputA = DiscretizeMean(outputA, granularity);
+        samplefileA << discretized_outputA << "\n";
+        double outputB = DPMean(valuesB, granularity,
+          implemented_epsilon, max_partitions, max_contributions, lower, upper);
+        double discretized_outputB = DiscretizeMean(outputB, granularity);
+        samplefileB << discretized_outputB << "\n";
+      }
+      samplefileA.close();
+      samplefileB.close();
+    }
+    else {
+      for (int i=0; i<number_of_samples; i++) {
+      double outputA = DPMean(valuesA, granularity,
```

Comment:
Shouldn't these both be DPLargeMean?

Reply:
So, the DPLargeMean is meant for really large values to prevent memory overload. In these situations, it was only the value of OutputB that necessitated DPLargeMean, OutputA could be constructed using DPMean.